pkg/sentry/fsimpl/fuse/fusefs.go

```diff
@@ -222,13 +271,281 @@ func (fs *filesystem) newInode(creds *auth.Credentials, mode linux.FileMode) *ke
 
 // Open implements kernfs.Inode.Open.
 func (i *inode) Open(ctx context.Context, rp *vfs.ResolvingPath, vfsd *vfs.Dentry, opts vfs.OpenOptions) (*vfs.FileDescription, error) {
-	fd, err := kernfs.NewGenericDirectoryFD(rp.Mount(), vfsd, &i.OrderedChildren, &i.locks, &opts, kernfs.GenericDirectoryFDOptions{
-		SeekEnd: kernfs.SeekEndStaticEntries,
-	})
+	isDir := i.InodeAttrs.Mode().IsDir()
+	if !isDir && opts.Mode.IsDir() {
```

Comment:
Can we test `isDir != opts.Mode.IsDir()`? Since if it's `isDIr && !opts.Mode.IsDir()` then it's still an error, right?

Reply:
Nope, if the file is a directory but O_DIRECTORY not passed, there won't be an error